### ISSUE TITLE
health_check default initial_delay is too low, timeout too high

### ISSUE THREAD

**Issue by matti** [kontena/kontena#3008]
Many containers run some migrations / prepration steps - the default `intial_delay: 10` causes weird problems - it would be safer to set it to 60 or so?

While at it, the default timeout at 10s is kinda high?

**Comment by SpComb** [comment]
Agreed, I've seen situations where the healthcheck restarts actually cause more problems than they solve, specifically regarding the `initial_delay` during container startup. The problem is that the agent healthcheck worker is not coupled into the `ServicePod` lifecycle, so the `initial_delay` starts counting up from the Docker container startup, so it overlaps with everything including the `weaveexec attach`, the `post_start` hooks and any `wait_for_port`.

The healthchecks should only start after all of those complete and the container is actually running, and then the `initial_delay` should be way less relevant at that point. That needs better integration between the agent `ServicePodWorker` and the other actors that are currently coupled via the Docker container events.
